lib/js/tests/gl/canvas2d_test.js

```diff
@@ -30,9 +32,35 @@ ctx.lineJoin = Canvas2dRe.LineJoin[/* round */0];
 
 ctx.miterLimit = 10;
 
-ctx.strokeStyle = "red";
+Canvas2dRe.setStrokeStyle(ctx, /* String */Block.__(0, ["red"]));
 
-ctx.fillStyle = "red";
+Canvas2dRe.setFillStyle(ctx, /* String */Block.__(0, ["red"]));
+
+var match = Canvas2dRe.fillStyle(ctx);
+
+switch (match.tag | 0) {
+  case 1 : 
+      match[0].addColorStop(0.0, "red");
+      break;
+  case 0 : 
+  case 2 : 
+      break;
+  
+}
+
+var match$1 = Canvas2dRe.strokeStyle(ctx);
+
+switch (match$1.tag | 0) {
+  case 0 : 
+      console.log(match$1[0]);
```

Comment:
You can see here how the GADT returned is an array with a tag property to switch on, and that the actual value is extracted by indexing into the array.